### ISSUE TITLE
Remove XML property, use attributes instead

### ISSUE THREAD

**Issue by brianvanderburg2** [Secretchronicles/TSC#354]
When we go to break compatibility, we should eliminate the XML property system that is currently used and instead use attributes.  Load/save code for individual items could also use subnodes if needed:

Instead of:
```xml
<sprite>
    <property name="type" value="1" />
    <property name="posx" value="100" />
    <property name="posy" value="100" />
</sprite>
```

We should use
```xml
<sprite type="1" posx="100" posy="100" />
```

Affects #179, #186 as well

**Comment by datahead8888** [comment]
It also affects: #169

We should probably decide if we want to first have a TSC 3.X release with a converter script and then introduce some of the newer data format conversions in subsequent releases (making use of the chain of conversion scripts) or introduce all the planned data conversions at once and have a more complex first 3.X release data conversion script (but less steps in the chain).

**Comment by Luiji** [comment]
I definitely think we should make this change before the 2.0 release. We want to reduce the amount of backwards-compatibility code necessary as much as possible, so the more changes to the format we do in bulk, the less versions of the level format will exist.

**Comment by datahead8888** [comment]
> I definitely think we should make this change before the 2.0 release.

Do you mean the first 3.0 release?  If I understood Brian correctly, this would break compatibility altogether if introduced in version 2.0, where we've been maintaining this compatibility so far.

You have a good point -- we probably don't want N releases dragging out the format changes when we finally break compatibility.  That would drag the format changes out forever.

**Comment by Quintus** [comment]
Hmmmm. There has been a lot of fuss about the 2.0 vs. 3.0 backward compatibility stuff. IMO quite late, but I wil bring the subject up on the mailinglist as it affects much more than just this ticket.

Vale,
Quintus

**Comment by Luiji** [comment]
@datahead8888, @Quintus This wouldn't break backwards-compatibility so long as we continue to consider smclvl and tsclvl different formats, with smclvl automatically importing.

**Comment by datahead8888** [comment]
> @datahead8888, @Quintus This wouldn't break backwards-compatibility so long as we continue to consider smclvl and tsclvl different formats, with smclvl automatically importing.

This is a very interesting idea, and the easiest time to take advantage of this is now for release 2.0 (since it is a major release and since this is when the file extension changes).  Once release 2.0 is finished, we really can't change the format of .tsclvl files without breaking levels, requiring us to wait until the next major release.

However, if we introduce new format changes in tsclvl files, the question is - how many format changes will we introduce?  Any format change (even just one) has a chance of introducing bugs in the code, which will slow our already long running release cycle down.

I know Quintus sent out an email to the mailing list about reverse compatibility and how long the release 2.X series should last.  I had some new thoughts which will take time to type up versus other stuff going on for school and TSC itself.  I don't think we can break compatibility in release 2.0 this late and that we need reverse compatibility in order to bring in the SMC users to our new game fork.  I'll have to wait on the other suggestions until I can type them up.

**Comment by Quintus** [comment]
Superseded by #380.
